lib/hyrax/configuration.rb

```diff
@@ -109,19 +109,41 @@ def register_roles
     attr_reader :analytics
     def analytics?
       @analytics ||=
-        ActiveModel::Type::Boolean.new.cast(ENV.fetch('HYRAX_ANALYTICS', false))
+        ActiveModel::Type::Boolean.new.cast(ENV.fetch('HYRAX_ANALYTICS', 'false'))
     end
 
-    attr_writer :google_analytics_id
-    def google_analytics_id
-      @google_analytics_id ||= nil
+    # Currently supports 'google' or 'matomo'
+    # google is default for backward compatability
+    attr_writer :analytics_provider
+    def analytics_provider
+      @analytics_provider ||=
+        ENV.fetch('HYRAX_ANALYTICS_PROVIDER', 'google')
+    end
+
+    # Date you wish to start collecting analytics for
+    # This is used to compute the "all-time" metrics
+    # Set this in your .env file (format is YYYY-MM-DD)
+    attr_writer :analytics_start_date
+    def analytics_start_date
+      @analytics_start_date ||=
+        ENV.fetch('ANALYTICS_START_DATE', Time.zone.today - 1.year)
```

Comment:
this looks like it will return a `Date` in the fallback case, but a `String`(?) when it's configured via ENV. is that okay?

Reply:
yes